### PR TITLE
[DEVOPS-2931] Upgrade to Node 18

### PR DESCRIPTION
**Pull request by gotrevor-notarize**
:ticket: [JIRA ticket: DEVOPS-2931](https://notarize.atlassian.net/browse/DEVOPS-2931) :ticket:

### CODE REVIEW

**Comment by dairyisscary** [comment]
what the heck? why is `node_modules` checked in? is that a requirement of github actions?

* thoughts on removing this directory from the repo?

File: bin/rebuild-for-deploy.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/usr/bin/env bash
+set -ex
+script_dir=$(cd -P -- "$(dirname -- "$0")" && pwd -P)
+cd $script_dir || exit 1
+cd .. || exit 1
+
+rm -rf node_modules
+rm -f package-lock.json
+npm install -omit=dev
```
Comment (dairyisscary) [inline bin/rebuild-for-deploy.sh:9]: we're reinstalling node_modules anyway? even though its checked in?

File: action.yml [hunk 2]
```diff
@@ -5,5 +5,5 @@ inputs:
     description: 'The GITHUB_TOKEN secret'
     required: true
 runs:
-  using: 'node12'
+  using: 'node16'
```
Comment (dairyisscary) [inline action.yml:8]: node 16? or 18?

File: bin/rebuild-for-deploy.sh (hunk 1 above)
Comment (gotrevor-notarize) [reply bin/rebuild-for-deploy.sh:9]: This is a helper file for developers.  There is no CI/CD.
I took a light touch to this update.
The README make a half-hearted attempt to explain this.

File: action.yml (hunk 2 above)
Comment (gotrevor-notarize) [reply action.yml:8]: Github actions don't support node16.  You'll see node18 in the commit history :)

**Review by dairyisscary — APPROVED** [review]
i can't even review this lol. my browser crashes on the diff 😭 

seems good though

**Comment by gotrevor-notarize** [comment]
@dairyisscary 
> what the heck? why is node_modules checked in? is that a requirement of github actions?
> thoughts on removing this directory from the repo?

Github actions requires these to be available somewhere.
The original instructions from GitHub whence this originated suggests this approach.
They've since advocated other options, but still, even those include checking in build artifacts.

I decided to stick with a light touch here, and just keep checking in the node_modules.

**Pull request merged**